charmhub_lp_tools/charm_project.py

```diff
@@ -444,12 +445,45 @@ def get_revisions_for_bases(self,
                 if delete:
                     del revisions_[all_arch]
         # now just keep the highest revision for each arch.
-        highest_revisions = collections.defaultdict(int)
+        highest_revisions: Dict[str, int] = {}
         for k, v in revisions_.items():
             highest_revisions[k] = list(sorted(revisions_[k]))[-1]
-        # and collect the final set of revisions to keep.
-        revisions = set(v for v in highest_revisions.values())
-        return revisions
+        return highest_revisions
+
+    @staticmethod
+    def revisions_from_revisions_by_arch(
+        arch_revisions: Dict[str, int]
+    ) -> List[int]:
+        """Return a sorted list of revisions from arch_revisions.
+
+        :param arch_revisions: dictionary of arch -> revision_id
+        :return: sorted list of revision_id.
+        """
+        revisions = set(v for v in arch_revisions.values())
+        return sorted(revisions)
+
+    @staticmethod
+    def str_revisions_by_arch(
+        arch_revisions: Dict[str, int]
+    ) -> str:
+        """Provide a str representation of the arch_revisions dict."""
+        return ','.join(f"{a}: {r}" for a, r in arch_revisions.items())
+
+    def get_revisions_for_bases(self,
+                                bases: List[str],
+                                arch: Optional[str] = None,
+                                ignore_arches: Optional[List[str]] = None,
+                                ) -> List[int]:
```

Comment:
There is - https://pypi.org/project/orderedset/ - but let's better keep it as it is instead of pulling in yet-another-dependency for a small value to the codebase.